api/src/R/checks.R

```diff
@@ -0,0 +1,70 @@
+#' Generic functions for checking assertions on response nodes.
+#'
+#' @param x Typically single response node, may be a numeric or character
+#' vector, or a list of such objects.
+#' @param options named vector (or list) of allowed options for a response
+check = function(x, ...) UseMethod("check")
+
+check.default = function(x, ...) return(invisible(NULL)) # nolint
+
+check.technical_skills = function(x, options, ...) { # nolint
+  assertthat::assert_that(is.character(x))
+  assertthat::assert_that(length(x) == 10)
```

Comment:
A lot of assertions in this file seem to be comparing with numbers, which, to me indicates that there must be a business logic reason for the fixed values. Should these be defined as constants somewhere?

Reply:
Thanks @sparkhi: in the particular example I see there I know that we currently have 10 skills listed for input in the technical skills question, so if we changed that at some point (not impossible as currently its based on the DigCurv framework and DPC  have recently introduced a new skills framework) then the number might change. I guess that if the number is only used in one place there may not be much advantage in setting a constant versus just declaring them in the test.